ddl/ddl.go

```diff
@@ -1547,36 +1545,43 @@ func processJobs(process func(*sess.Session, *model.Job, model.AdminCommandOpera
 			if !ok {
 				logutil.BgLogger().Debug("Job ID from meta is not consistent with requested job id,",
 					zap.Int64("fetched job ID", job.ID))
-				errs[i] = dbterror.ErrInvalidDDLJob.GenWithStackByArgs(job.ID)
+				jobErrs[i] = dbterror.ErrInvalidDDLJob.GenWithStackByArgs(job.ID)
 				continue
 			}
 			delete(jobMap, job.ID)
 
 			err = process(ns, job, byWho)
 			if err != nil {
-				errs[i] = err
+				jobErrs[i] = err
 				continue
 			}
 
 			err = updateDDLJob2Table(ns, job, true)
 			if err != nil {
-				errs[i] = err
+				jobErrs[i] = err
 				continue
 			}
 		}
 
+		failpoint.Inject("mockCommitFailedOnDDLCommand", func(val failpoint.Value) {
```

Comment:
It seems we can use `mockCommitErrorInNewTxn` instead of this one, so we can remove this one.

Reply:
It doesn't work though.

Reply:
It feels like `TestCancelJobWriteConflict` is working. Why doesn't pause work?

Reply:
No.

It works because of `mockFailedCommandOnConcurencyDDL`, but not `mockCommitErrorInNewTxn`.

Reply:
Or, could we redesign the whole test here later on? The scattered 'fail point' may need to be reviewed as a whole.

`mockCommitErrorInNewTxn` is injected in `RunInNewTxn` while `mockCommitError` is injected in `doCommit`, and they look like two different transaction mechanism. Here in `cancel`, `pause` and `resume`, they are using the `doCommit` but not `RunInNewTxn` which may need to be reviewed too.